Importing Libraries

In [14]:
import os
import pandas as pd
print(pd.__version__)

2.3.0


Load Dataset

In [15]:
file_path = os.path.join(
    os.getcwd(), "..", "data", "raw", "Pakistan Largest Ecommerce Dataset.csv"
)
data = pd.read_csv(file_path)

C:\Users\malip\AppData\Local\Temp\ipykernel_16360\680866094.py:4: DtypeWarning: Columns (1,2,3,7,8,9,11,12,13,14,17,18,19) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv(file_path)


In [16]:
data.head(2)

item_id    status created_at  \
0  211131.0  complete   7/1/2016   
1  211133.0  canceled   7/1/2016   

                                                 sku   price  qty_ordered  \
0                                  kreations_YI 06-L  1950.0          1.0   
1  kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Bo...   240.0          1.0   

   grand_total increment_id    category_name_1 sales_commission_code  ...  \
0       1950.0    100147443    Women's Fashion                    \N  ...   
1        240.0    100147444  Beauty & Grooming                    \N  ...   

   Month Customer Since     M-Y    FY Customer ID  Unnamed: 21  Unnamed: 22  \
0    7.0         2016-7  7-2016  FY17         1.0          NaN          NaN   
1    7.0         2016-7  7-2016  FY17         2.0          NaN          NaN   

  Unnamed: 23 Unnamed: 24 Unnamed: 25  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  

[2 rows x 26 columns]

In [17]:
data.describe()

item_id         price    qty_ordered   grand_total  \
count  584524.000000  5.845240e+05  584524.000000  5.845240e+05   
mean   565667.074218  6.348748e+03       1.296388  8.530619e+03   
std    200121.173648  1.494927e+04       3.996061  6.132081e+04   
min    211131.000000  0.000000e+00       1.000000 -1.594000e+03   
25%    395000.750000  3.600000e+02       1.000000  9.450000e+02   
50%    568424.500000  8.990000e+02       1.000000  1.960400e+03   
75%    739106.250000  4.070000e+03       1.000000  6.999000e+03   
max    905208.000000  1.012626e+06    1000.000000  1.788800e+07   

       discount_amount           Year          Month    Customer ID  \
count    584524.000000  584524.000000  584524.000000  584513.000000   
mean        499.492775    2017.044115       7.167654   45790.511965   
std        1506.943046       0.707355       3.486305   34414.962389   
min        -599.500000    2016.000000       1.000000       1.000000   
25%           0.000000    2017.000000       4.000000   13516.000000   
50%           0.000000    2017.000000       7.000000   42856.000000   
75%         160.500000    2018.000000      11.000000   73536.000000   
max       90300.000000    2018.000000      12.000000  115326.000000   

       Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  Unnamed: 25  
count          0.0          0.0          0.0          0.0          0.0  
mean           NaN          NaN          NaN          NaN          NaN  
std            NaN          NaN          NaN          NaN          NaN  
min            NaN          NaN          NaN          NaN          NaN  
25%            NaN          NaN          NaN          NaN          NaN  
50%            NaN          NaN          NaN          NaN          NaN  
75%            NaN          NaN          NaN          NaN          NaN  
max            NaN          NaN          NaN          NaN          NaN

In [18]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 26 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   item_id                584524 non-null  float64
 1   status                 584509 non-null  object 
 2   created_at             584524 non-null  object 
 3   sku                    584504 non-null  object 
 4   price                  584524 non-null  float64
 5   qty_ordered            584524 non-null  float64
 6   grand_total            584524 non-null  float64
 7   increment_id           584524 non-null  object 
 8   category_name_1        584360 non-null  object 
 9   sales_commission_code  447346 non-null  object 
 10  discount_amount        584524 non-null  float64
 11  payment_method         584524 non-null  object 
 12  Working Date           584524 non-null  object 
 13  BI Status              584524 non-null  object 
 14   MV                    584524 non-

In [19]:
df = data.copy()

Drop NAN Columns

In [20]:
df = df.drop(["Unnamed: 21","Unnamed: 22",	"Unnamed: 23",	"Unnamed: 24",	"Unnamed: 25"],axis=1)

Checking Missing Values

In [21]:
df.isnull().sum()

item_id                  464051
status                   464066
created_at               464051
sku                      464071
price                    464051
qty_ordered              464051
grand_total              464051
increment_id             464051
category_name_1          464215
sales_commission_code    601229
discount_amount          464051
payment_method           464051
Working Date             464051
BI Status                464051
 MV                      464051
Year                     464051
Month                    464051
Customer Since           464062
M-Y                      464051
FY                       464051
Customer ID              464062
dtype: int64

In [22]:
df.notnull().sum()

item_id                  584524
status                   584509
created_at               584524
sku                      584504
price                    584524
qty_ordered              584524
grand_total              584524
increment_id             584524
category_name_1          584360
sales_commission_code    447346
discount_amount          584524
payment_method           584524
Working Date             584524
BI Status                584524
 MV                      584524
Year                     584524
Month                    584524
Customer Since           584513
M-Y                      584524
FY                       584524
Customer ID              584513
dtype: int64

In [23]:
df = df.dropna(axis=0)

In [24]:
df.isnull().sum()

item_id                  0
status                   0
created_at               0
sku                      0
price                    0
qty_ordered              0
grand_total              0
increment_id             0
category_name_1          0
sales_commission_code    0
discount_amount          0
payment_method           0
Working Date             0
BI Status                0
 MV                      0
Year                     0
Month                    0
Customer Since           0
M-Y                      0
FY                       0
Customer ID              0
dtype: int64

In [25]:
df.notnull().sum()

item_id                  447326
status                   447326
created_at               447326
sku                      447326
price                    447326
qty_ordered              447326
grand_total              447326
increment_id             447326
category_name_1          447326
sales_commission_code    447326
discount_amount          447326
payment_method           447326
Working Date             447326
BI Status                447326
 MV                      447326
Year                     447326
Month                    447326
Customer Since           447326
M-Y                      447326
FY                       447326
Customer ID              447326
dtype: int64

In [26]:
df.head(2)

item_id    status created_at  \
0  211131.0  complete   7/1/2016   
1  211133.0  canceled   7/1/2016   

                                                 sku   price  qty_ordered  \
0                                  kreations_YI 06-L  1950.0          1.0   
1  kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Bo...   240.0          1.0   

   grand_total increment_id    category_name_1 sales_commission_code  ...  \
0       1950.0    100147443    Women's Fashion                    \N  ...   
1        240.0    100147444  Beauty & Grooming                    \N  ...   

   payment_method Working Date BI Status      MV     Year  Month  \
0             cod     7/1/2016     #REF!   1,950   2016.0    7.0   
1             cod     7/1/2016     Gross     240   2016.0    7.0   

   Customer Since     M-Y    FY Customer ID  
0          2016-7  7-2016  FY17         1.0  
1          2016-7  7-2016  FY17         2.0  

[2 rows x 21 columns]

In [27]:
df["sales_commission_code"].value_counts()

sales_commission_code
\N              338983
40968             2710
C-LHW-50074       2434
C-RWP-31924       1719
cisb30211         1613
                 ...  
C--PEW-51830         1
C-FSD-105093         1
Combed-008           1
C-GUJ100603          1
C-43409-SGD          1
Name: count, Length: 7225, dtype: int64

In [28]:
print(df["sales_commission_code"].nunique())
print(df["sales_commission_code"].unique())

print(df["BI Status"].nunique())
print(df["BI Status"].unique())

7225
['\\N' 'R-FSD-52352' '105259' ... 'GRS-4090974-1' 'GRS-4090975-1'
 '100559728']
4
['#REF!' 'Gross' 'Net' 'Valid']


Renaming Column

In [29]:
df.rename(columns={'category_name_1':'Category'},inplace=True)
df.columns

Index(['item_id', 'status', 'created_at', 'sku', 'price', 'qty_ordered',
       'grand_total', 'increment_id', 'Category', 'sales_commission_code',
       'discount_amount', 'payment_method', 'Working Date', 'BI Status',
       ' MV ', 'Year', 'Month', 'Customer Since', 'M-Y', 'FY', 'Customer ID'],
      dtype='object')

In [30]:
print(df["Category"].nunique())
print(df["Category"].unique())

16
["Women's Fashion" 'Beauty & Grooming' 'Soghaat' 'Mobiles & Tablets'
 'Appliances' 'Home & Living' "Men's Fashion" 'Kids & Baby' '\\N' 'Others'
 'Entertainment' 'Computing' 'Superstore' 'Health & Sports' 'Books'
 'School & Education']


Checking Completed Orders

In [31]:
completed_orders = df[df["status"] == "complete"]
completed_orders.head(2)

item_id    status created_at                sku   price  qty_ordered  \
0  211131.0  complete   7/1/2016  kreations_YI 06-L  1950.0          1.0   
3  211135.0  complete   7/1/2016     kcc_krone deal   360.0          1.0   

   grand_total increment_id           Category sales_commission_code  ...  \
0       1950.0    100147443    Women's Fashion                    \N  ...   
3         60.0    100147446  Beauty & Grooming           R-FSD-52352  ...   

   payment_method Working Date BI Status      MV     Year  Month  \
0             cod     7/1/2016     #REF!   1,950   2016.0    7.0   
3             cod     7/1/2016       Net     360   2016.0    7.0   

   Customer Since     M-Y    FY Customer ID  
0          2016-7  7-2016  FY17         1.0  
3          2016-7  7-2016  FY17         4.0  

[2 rows x 21 columns]

In [32]:
category_complete = completed_orders["Category"].value_counts()
category_complete.columns = ['category_name','completed_count']
category_complete .head(5)


Category
Mobiles & Tablets    37118
Men's Fashion        35227
Women's Fashion      20289
Soghaat              19138
Beauty & Grooming    18766
Name: count, dtype: int64

In [33]:
top_category = category_complete.head(1)
print(f"The best selling Category is :{top_category}:")

The best selling Category is :Category
Mobiles & Tablets    37118
Name: count, dtype: int64:


The best selling Category is Mobiles & Tablets with completed orders of 37118.

In [34]:
df.to_csv("cleaned_dataset.csv", index=False)
df.to_pickle("cleaned_dataset.pkl")
